Generate project file from tmux session

pczarn opened the pull request:
Implements #61
* [x] windows and panes
* [x] layout
* [x] path
* [ ] commands
* [ ] history

See also
http://manpages.ubuntu.com/manpages/precise/en/man1/tmux.1.html#contenttoc9
http://blog.edsantiago.com/articles/tmux-session-preserve/
http://zerokspot.com/weblog/2012/05/20/tmux-split-window-and-pwd/
https://github.com/mislav/dotfiles/blob/master/bin/tmux-session

DeepAnchor commented:
this is a very useful feature, any reason to not merge this?

adamstrickland commented:
I agree, this looks useful.  However, I'd suggest moving this to `Tmuxinator::Cli#new_from_session`.  Also, some tests need to be added (not sure if `spec/lib/tmuxinator/cli_spec.rb` was around when this code was originally written, but it is now.

ianks commented:
@adamstrickland it is about 2 years late for code nits like that IMO. Low chance the contributor will make it back to fix those. Might as well merge it then refactor however is necessary.

pczarn commented:
@ianks Yes, but I can make these fixes.

adamstrickland commented:
@ianks I agree, the chance is low, whether those are "nits" or not.  Speaking for myself, I don't like the idea of adding technical debt (one of those "nits" of which you speak), when in the intervening 2 years only but one or two contributors have been so inspired as to even express interest in furthering this functionality.

Regardless, since @DeepAnchor recently expressed such interest, perhaps they would be so inspired.  Or yourself, even, rather than leaving it to the original contributor.  But thank you for your opinion.

adamstrickland commented:
Thanks @pczarn.  Sorry this wasn't incorporated way back when :-/

ianks commented:
Sorry if that came off wrong @adamstrickland, I just don't like seeing when a contributor puts a lot of time and effort into adding functionality to a project, just to have it ignored. I think its rather unfair. I'm happy it is working out this time though :smile:

adamstrickland commented:
@ianks Fair enough.  I agree, having been on the other end of a similar situation, it is both frustrating and unfair, especially well after the fact.  Which is why I responded now (my first chance) that @DeepAnchor raised the question.  That being said, the [contributing guide](https://github.com/tmuxinator/tmuxinator/blob/master/CONTRIBUTING.md#pull-requests) pretty clearly states that tests are required in PRs...

Regardless, like you said, I'm happy this looks to be working out this time.  I just got back to my computer after some time off, so I personally am backed up for a few days, but I'll take a look myself at adding a test or two when I get a chance.  Hopefully we can get this :ship:ed ASAP.

pczarn commented:
I updated the code and added a test. I see a small problem which is adding low-level logic in `Cli`. Ideally parts of this code would be moved to some other files, but I'm not sure how to proceed.

pczarn commented:
Could you reopen? This PR was broken by the large difference in commit history.

J3RN commented:
@pczarn Unfortunately, it doesn't look like I can reopen this one, since it says that there are no commits between your `serialize-session` branch and our `0.7.0-wip` branch. It seems the large difference in commit history was simply because this pull request was opened against that `0.7.0-wip` branch, which is very out-of-date.

Could you push up your changes to your `serialize-session` branch again open a new pull-request against our `master` branch?

pczarn commented:
Done, the new PR is #426
